eng/common/pipelines/templates/jobs/archetype-spec-gen-sdk.yml

```diff
@@ -0,0 +1,201 @@
+          cd $(SpecRepoDirectory)
+          default_commit=$(git rev-parse HEAD)
+          echo "##vso[task.setvariable variable=SpecRepoCommit]$default_commit"
+          echo "SpecRepoCommit variable set to default commit: $default_commit"
+        else
+          echo "##vso[task.setvariable variable=SpecRepoCommit]${{ parameters.SpecRepoCommit }}"
+          echo "SpecRepoCommit variable set to: ${{ parameters.SpecRepoCommit }}"
+        fi
+      displayName: 'Set SpecRepoCommit variable'
+
+    - task: NodeTool@0
+      inputs:
+        versionSpec: $(NodeVersion)
+      displayName: 'Install Node.js'
+
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: $(PythonVersion)
+
+    - script: |
+        npm install -g @azure-tools/spec-gen-sdk@$(SpecGenSdkVersion)
+      displayName: 'Install spec-gen-sdk'
+
+    - script: |
+        optional_params=""
+        sdk_gen_info="sdk generation from Config : "
+
+        if [ "${{ parameters.ConfigType }}" = "TypeSpec" ]; then
+          optional_params="$optional_params --tsp-config-relative-path ${{ parameters.ConfigPath }}"
+          sdk_gen_info="$sdk_gen_info '${{ parameters.ConfigPath }}',"
+        else
+          optional_params="$optional_params --readme-relative-path ${{ parameters.ConfigPath }}"
+          sdk_gen_info="$sdk_gen_info '${{ parameters.ConfigPath }}',"
+        fi
+
+        sdk_gen_info="$sdk_gen_info and CommitSHA: '$(SpecRepoCommit)', in SpecRepo: '${{ parameters.SpecRepoUrl }}'"
+        echo "##vso[task.setvariable variable=GeneratedSDKInformation]$sdk_gen_info"
+        echo "Generated SDK Information : $sdk_gen_info"
+
+        spec-gen-sdk \
+          --scp "$(SpecRepoDirectory)" \
+          --sdp "$(SdkRepoDirectory)" \
+          --wf "$(System.DefaultWorkingDirectory)" \
+          -l "$(SdkRepoNameForSpecGen)" \
+          -c "$(SpecRepoCommit)" \
+          -t $true \
+          $optional_params
+      displayName: 'Generate SDK'
+
+    - task: PublishPipelineArtifact@1
+      displayName: Publish SDK Artifact to Pipeline Artifacts
+      inputs:
+        artifactName: $(sdkArtifactName)
+        targetPath: "$(System.DefaultWorkingDirectory)/generatedSdkArtifacts"
+
+    - ${{ if eq(parameters.SkipPullRequestCreation, false) }}: 
+      - template: /eng/common/pipelines/templates/steps/git-push-changes.yml
+        parameters:
+          BaseRepoBranch: $(PrBranch)-$(Build.BuildId)
+          BaseRepoOwner: azure-sdk
+          CommitMsg: $(GeneratedSDKInformation)
+          TargetRepoOwner: $(SdkRepoOwner)
+          TargetRepoName: $(SdkRepoName)
+          PushArgs: "--force"
+          WorkingDirectory: $(SdkRepoDirectory)
+          ScriptDirectory: $(SdkRepoDirectory)/eng/common/scripts
+
+      - task: PowerShell@2
+        displayName: Create pull request
+        condition: and(succeeded(), eq('${{ parameters.SkipPullRequestCreation }}', false), eq(variables['HasChanges'], 'true'))
```

Comment:
You shouldn't need the SkipPullRequestCreation condition here given it is part of the template condition.